azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AADOAuth2AutoConfiguration.java

```diff
@@ -22,18 +30,36 @@
 @PropertySource(value = "classpath:serviceEndpoints.properties")
 @EnableConfigurationProperties({AADAuthenticationProperties.class, ServiceEndpointsProperties.class})
 public class AADOAuth2AutoConfiguration {
-    private AADAuthenticationProperties aadAuthProps;
-    private ServiceEndpointsProperties serviceEndpointsProps;
+
+    private final TelemetryProxy telemetryProxy;
+
+    private final AADAuthenticationProperties aadAuthProps;
+
+    private final ServiceEndpointsProperties serviceEndpointsProps;
 
     public AADOAuth2AutoConfiguration(AADAuthenticationProperties aadAuthProperties,
-                                      ServiceEndpointsProperties serviceEndpointsProps) {
+                                      ServiceEndpointsProperties serviceEndpointsProps,
+                                      TelemetryProxy telemetryProxy) {
         this.aadAuthProps = aadAuthProperties;
         this.serviceEndpointsProps = serviceEndpointsProps;
+        this.telemetryProxy = telemetryProxy;
+
+        trackCustomEvent();
     }
 
     @Bean
     @ConditionalOnProperty(prefix = "azure.activedirectory", value = "active-directory-groups")
     public OAuth2UserService<OidcUserRequest, OidcUser> oidcUserService() {
         return new AADOAuth2UserService(aadAuthProps, serviceEndpointsProps);
     }
+
+    private void trackCustomEvent() {
```

Comment:
Can we move this `trackCustomEvent` as some shared function? Seems a little duplicate.

Reply:
Yes, I try a little for less duplicated, but it looks I can do more.

Reply:
keep this for now, as different service may have different metrics, like storage has `hashedAccountName`, and `b2c` has `tenantName`